### PR TITLE
Bonnie 1503

### PR DESCRIPTION
**Pull request by mayerm94**
Change unit->units in several data criteria types to match data coming from patient_builder.rb in bonnie. Add tests that use entry data coming from patient_builder in bonnie grabbed in the console during the initialization of the data criteria.

Pull requests into CQL QDM Patient API require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1503
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered.
    * In rare situations, this may not be possible or applicable to a PR. In those situations:
        1. Note why this could not be done or is not applicable here:
        2. Add TODOs in the code noting that it requires a test
        3. Add a JIRA task to add the test and link it here:
- [x] Automated regression test(s) pass One regression on the MPR account, dosage units were not seen before, now that they are an invalid UCUM unit 'mcg' error is thrown

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links: NA
- [x] Justification for using JIRA tests: NA
- [x] JIRA tests have been added to sprint NA

**Reviewer 1:**

Name: @losborne
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass N/A
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree N/A

**Reviewer 2:**

Name: @hossenlopp
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests: N/A
- [x] JIRA tests have been run and pass N/A
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree N/A

### CODE REVIEW

File: app/assets/javascripts/datatypes/medication.js.coffee [hunk 1]
```diff
@@ -72,7 +72,7 @@ class CQL_QDM.MedicationActive extends CQL_QDM.QDMDatatype
   ###
   supply: ->
     if @_supply?
-      new cql.Quantity({unit: @_supply['unit'], value: @_supply['scalar']})
+      new cql.Quantity({unit: @_supply['units'], value: @_supply['scalar']})
```
Comment (hossenlopp) [inline app/assets/javascripts/datatypes/medication.js.coffee]: This line is not covered by unit tests.

File: app/assets/javascripts/datatypes/medication.js.coffee [hunk 2]
```diff
@@ -251,7 +251,7 @@ class CQL_QDM.MedicationDischarge extends CQL_QDM.QDMDatatype
...
```
Comment (hossenlopp) [inline app/assets/javascripts/datatypes/medication.js.coffee]: This line is not covered by unit tests.

File: app/assets/javascripts/datatypes/laboratorytest.js.coffee [hunk 3]
```diff
@@ -124,19 +124,19 @@ class CQL_QDM.LaboratoryTestPerformed extends CQL_QDM.QDMDatatype
     # According to documentation, this is assumed to be a 'Quantity'
     high = null
     low = null
-    if @_referenceRangeHigh?['unit']?
+    if @_referenceRangeHigh?['units']?
```
Comment (hossenlopp) [inline app/assets/javascripts/datatypes/laboratorytest.js.coffee]: The body of this if are not covered.

File: app/assets/javascripts/datatypes/laboratorytest.js.coffee [hunk 4]
```diff
@@ -124,19 +124,19 @@ class CQL_QDM.LaboratoryTestPerformed extends CQL_QDM.QDMDatatype
...
       high_obj =
         unit:
-          @_referenceRangeHigh['unit']
+          @_referenceRangeHigh['units']
         value:
-          @_referenceRangeHigh['value']
+          @_referenceRangeHigh['scalar']
       high = new cql.Quantity(high_obj)
-    if @_referenceRangeLow?['unit']?
+    if @_referenceRangeLow?['units']?
```
Comment (hossenlopp) [inline app/assets/javascripts/datatypes/laboratorytest.js.coffee]: The body of this if are not covered.

File: app/assets/javascripts/datatypes/laboratorytest.js.coffee [hunk 5]
```diff
@@ -124,19 +124,19 @@ class CQL_QDM.LaboratoryTestPerformed extends CQL_QDM.QDMDatatype
...
       low_obj =
         unit:
-          @_referenceRangeLow['unit']
+          @_referenceRangeLow['units']
         value:
-          @_referenceRangeLow['value']
+          @_referenceRangeLow['scalar']
       low = new cql.Quantity(low_obj)
     if low?
```
Comment (hossenlopp) [inline app/assets/javascripts/datatypes/laboratorytest.js.coffee]: The body of this if are not covered.

File: app/assets/javascripts/datatypes/medication.js.coffee (hunk 1 above)
Comment (hossenlopp) [reply app/assets/javascripts/datatypes/medication.js.coffee]: Covered now.

File: app/assets/javascripts/datatypes/medication.js.coffee (hunk 2 above)
Comment (hossenlopp) [reply app/assets/javascripts/datatypes/medication.js.coffee]: Covered now.
